automatic/git.install/tools/chocolateyInstall.ps1

```diff
@@ -97,19 +103,28 @@ if ($noAutoCrlf) {
   New-ItemProperty $installKey -Name "Inno Setup CodeFile: CRLF Option" -Value "CRLFCommitAsIs" -PropertyType "String" -Force | Out-Null
 }
 
-# Make our install work properly when running under SYSTEM account (Chef Cliet Service, Puppet Service, etc)
+if ($disableShellIntegration) {
+  $fileArgs = $fileArgs.replace('ext\shellhere', '')
+  $fileArgs = $fileArgs.replace('ext\guihere', '')
+  $fileArgs = $fileArgs.replace('ext', '')
+}
+
+# Make our install work properly when running under SYSTEM account (Chef Client Service, Puppet Service, etc)
 # Add other items to this if block or use $IsRunningUnderSystemAccount to adjust existing logic that needs changing
 $IsRunningUnderSystemAccount = ([System.Security.Principal.WindowsIdentity]::GetCurrent()).IsSystem
 If ($IsRunningUnderSystemAccount)
 {
   #strip out quicklaunch parameter as it causes a hang under SYSTEM account
-  $fileArgs = $fileArgs.replace('icons\quicklaunch,','')
+  $fileArgs = $fileArgs.replace('icons\quicklaunch', '')
   If ($fileArgs -inotlike "*/SUPPRESSMSGBOXES*")
   {
     $fileArgs = $fileArgs + ' /SUPPRESSMSGBOXES'
   }
 }
 
+$fileArgs = $fileArgs.replace(',+$|^,+', '')
+$fileArgs = $fileArgs.replace(',{2,}', ',')
+

```

Comment:
Everything else looks good, but I am unclear on what this change is about. Can you explain this a bit?